fmt/printf.h

```diff
@@ -463,6 +472,23 @@ void PrintfFormatter<Char, AF>::format(BasicCStringRef<Char> format_str) {
     if (!*s)
       FMT_THROW(FormatError("invalid format string"));
     spec.type_ = static_cast<char>(*s++);
+
+    if (spec.type_ == 's' || spec.type_ == 'S') {
+        if (arg.type == Arg::POINTER) {
+            spec.type_ = 'p';
+        } else if (arg.type == Arg::CHAR) {
+            spec.type_ = 'c';
+        } else if (arg.type == Arg::BOOL) {
+            // leave bool alone
+        } else if (arg.type <= Arg::LAST_INTEGER_TYPE) {
+            // all other integer types
+            spec.type_ = 'd';
+        } else if (arg.type <= Arg::LAST_NUMERIC_TYPE) {
+            // all floating point types
+            spec.type_ = 'f';
+        }
+    }
```

Comment:
Please convert this if-else chain into a visitor and merge with the similar logic under for integers right below.